### PR TITLE
Linux - Scripting - Module - closes #552

### LINKED ISSUE #552: Linux - Scripting Module

### PR DESCRIPTION
**Pull request by BenHesketh21**

### CODE REVIEW

File: topics/linux/modules/scripting/README.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+## Overview
+Simply put a script is a file that we can execute. 
+The file would contain text in the form of commands and execute the commands one after another.
+
+We need to remember to make the command **executable**.
+
+## Tutorial
+
+We create a script file with the file extension `.sh`
+
```
Comment (tvaidotas) [inline topics/linux/modules/scripting/README.md]:
```suggestion


Execute the following command to create a script file called `example.sh`

```

File: topics/linux/modules/scripting/README.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+```bash
+touch example.sh
+```
+You can use `#` for comments.
+```bash
+whoami
+# Everything after the hash will be ignored
+ls
+```
+We need to specify where we want the file to be executed, we do this at the top of the file with this line to run the file as a bash script.
```
Comment (tvaidotas) [inline topics/linux/modules/scripting/README.md]:
```suggestion
We need to specify where we want the file to be executed, we do this at the top of the file with this line to run the file as a bash script. This line is also knows as `shebang`.
```

File: topics/linux/modules/scripting/README.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+```bash
+#! /bin/bash
+```
+Then we add the commands we want the script to execute to the file.
```
Comment (tvaidotas) [inline topics/linux/modules/scripting/README.md]:
```suggestion
Then we add the commands we want the script to execute below the shebang line. In the example below the first command `whoami` will print out the username of the current session, and the second command will list all the files within the directory.
```

File: topics/linux/modules/scripting/README.md [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+```bash
+#! /bin/bash
+
+whoami
+ls
+```
+We then exit the file and make it executable with this command.
```
Comment (tvaidotas) [inline topics/linux/modules/scripting/README.md]:
```suggestion
We then save and exit the file. In order for the script to be executable you need to add the executable permission, the command below is specifically for this purpose.
```

File: topics/linux/modules/scripting/README.md [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+Then we can run the file with this command.
+```bash
+./example.sh
+```
+Or we can add the directory that the file resides in to the PATH, to make it exectuable from anywhere.
+```bash
+export PATH=$(pwd):$PATH
+```
+
+### Scripting Logic
```
Comment (tvaidotas) [inline topics/linux/modules/scripting/README.md]: Adding variables would be good

Comment (BenHesketh21) [reply topics/linux/modules/scripting/README.md]: I will add a section on variables

File: topics/linux/modules/scripting/README.md [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+	mkdir $(date "+%B-%Y")
+	cd $(date "+%B-%Y")
+fi
+touch ${name}$(date "+%d").md
+echo "## ${title} Date: $(date "+%D")" >> ${name}$(date "+%d").md
+vim ${name}$(date "+%d").md
+```
+</details>
+
+**Exercise 6**
```
Comment (tvaidotas) [inline topics/linux/modules/scripting/README.md]: The solution needs to be more thorough, you need to specify that a gcloud account is needed, installing gcloud sdk and then logging in, executing the script, need to select a project, then the region, and at that point I gave up as it seems too much hassle to finish solving this

File: topics/linux/modules/scripting/README.md (hunk 5 above)
Comment (BenHesketh21) [reply topics/linux/modules/scripting/README.md]: Variables are already covered in the files and directories module, should I still add it in?

File: topics/linux/modules/scripting/README.md [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+echo "Title?"
+read title
+cd ~
+if [ -d "$(date "+%B-%Y")" ]; then
+	cd $(date "+%B-%Y")
+else
+	mkdir $(date "+%B-%Y")
+	cd $(date "+%B-%Y")
+fi
+touch ${name}$(date "+%d").md
```
[the remark this answers is not in the document]
Comment (BenHesketh21) [reply topics/linux/modules/scripting/README.md]: I have added clarification in the explaination

File: topics/linux/modules/scripting/README.md (hunk 6 above)
Comment (BenHesketh21) [reply topics/linux/modules/scripting/README.md]: Deleted

File: topics/linux/modules/scripting/README.md (hunk 5 above)
Comment (tvaidotas) [reply topics/linux/modules/scripting/README.md]: in this domain it's a variable in bash rather than a system variable, so it would help having it here

**Pull request merged**